app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -1199,6 +1208,12 @@ BrowserWorld::DrawImmersive() {
       m.blitter->CancelFrame(surfaceHandle);
     }
     DrawLoadingAnimation();
+
+    if ((m.device->GetDeviceType() != device::OculusGo) &&
```

Comment:
Can we use `if (surfaceType == mozilla::gfx::VRLayerTextureType::LayerTextureType_ExternalVRSurface)`instead of checking by device?

Reply:
Or just always call  m.device->SetExternalSurfId(-1) here

Reply:
Also this call seems to be in the else branch which shows the loading animation. Should it be in the other branch or at the end of the if conditions? next to the end `m.device->EndFrame` I mean

Reply:
It looks like `DrawLoadingAnimation ()` is at the else branch of `if (state == ExternalVR::VRState::Rendering)`. I didn't change the logic of this part, and it seems make sense to let draw loading animation when it is not at the rendering mode.